packages/core/src/Script.ts

```diff
@@ -219,20 +219,11 @@ export class Script extends Component {
    */
   _handlingInValid(): void {
     const componentsManager = this.engine._componentsManager;
-    // Use "xxIndex !== -1" to project.
-    // Maybe call onDisable it is still not in script queue, for example write "entity.isActive = false" in onWake().
-    if (this._onUpdateIndex !== -1) {
-      componentsManager.removeOnUpdateScript(this);
-    }
-    if (this._onLateUpdateIndex !== -1) {
-      componentsManager.removeOnLateUpdateScript(this);
-    }
-    if (this._onPhysicsUpdateIndex !== -1) {
-      componentsManager.removeOnPhysicsUpdateScript(this);
-    }
-    if (this._entityScriptsIndex !== -1) {
-      this._entity._removeScript(this);
-    }
+    componentsManager.removeOnUpdateScript(this);
+    componentsManager.removeOnLateUpdateScript(this);
+    componentsManager.removeOnPhysicsUpdateScript(this);
```

Comment:
You don't need to consider the case of -1, is it guaranteed not to have?

Reply:
After this PR https://github.com/oasis-engine/engine/pull/1055 fixed, call onDisable before onEnable maybe never happen.

Reply:
> After this PR #1055 fixed, call onDisable before onEnable maybe never happen.

Even if repaired, the subscript of `onUpdate` may be -1.
```
if (this.onUpdate !== prototype.onUpdate) {
        componentsManager.addOnUpdateScript(this);
}
```

Reply:
done